src/txn/txn.c

```diff
@@ -552,6 +552,21 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
     if (cval.val)
         F_SET(txn, WT_TXN_TS_ROUND_READ);
 
+    /* Check if the read timestamp is allowed to be before the oldest timestamp. */
+    WT_RET(__wt_config_gets_def(session, cfg, "allow_read_timestamp_before_oldest", 0, &cval));
+    if (cval.val) {
+        if (F_ISSET(txn, WT_TXN_TS_ROUND_READ))
+            WT_RET_MSG(session, EINVAL,
+              "cannot specify roundup_timestamps.read and "
+              "allow_read_timestamp_before_oldest on the same transaction");
+        F_SET(txn, WT_TXN_TS_READ_BEFORE_OLDEST);
+    } else
+        /*
+         * We need to clear the flag here in case it was set in a previous iteration of transaction
+         * begin that exited due to an error, as such the flag may still be set.
+         */
+        F_CLR(txn, WT_TXN_TS_READ_BEFORE_OLDEST);
```

Comment:
This bit weirds me out. Does something need to happen in `__txn_destroy` (or whatever it's called)?
And how come we don't need to do this for the other flags?

Reply:
I _think_ its an issue for all the other flags too but generally we don't fail in `__wt_txn_begin` so no one has noticed, I only picked it up while writing the test. The function that clears the flags is `__wt_txn_release` which only gets called from commit and rollback.

Reply:
I'll write a test to see if the rounding flag for example doesn't get cleared, when we fail in begin to know for sure.

Reply:
So I can confirm that this is an issue for all transaction flags set in `__wt_txn_config` as such I've added an error handling section to the function.